internal/helm/repository/chart_repository.go

```diff
@@ -356,7 +363,7 @@ func (r *ChartRepository) LoadFromPath() error {
 // DownloadIndex attempts to download the chart repository index using
 // the Client and set Options, and writes the index to the given io.Writer.
 // It returns an url.Error if the URL failed to parse.
-func (r *ChartRepository) DownloadIndex(w io.Writer) (err error) {
+func (r *ChartRepository) DownloadIndex(w io.Writer, n int64) (err error) {
```

Comment:
```suggestion
func (r *ChartRepository) DownloadIndex(w io.Writer, maxSize int64) (err error) {
```

Also add maxSize to the function comment above please.